release.csx

```diff
@@ -0,0 +1,84 @@
+#r ".\packages\Octokit.0.26.0\lib\net45\Octokit.dll"
+#r ".\packages\system.runtime.interopservices.runtimeinformation\4.0.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll"
+
+using Octokit;
+
+const string repoOwner = "FakeItEasy";
+const string repoName = "FakeItEasy";
+const string remote = "upstream";
+
+var version = Args.FirstOrDefault();
+if (version == null)
+{
+    throw new Exception("No version supplied");
+}
+
+var releaseBranchName = $"release/{version}";
+
+var token = ReadTokenFromNetrc();
+
+var gitHubClient = new GitHubClient(new ProductHeaderValue("FakeItEasy-build-scripts"));
+gitHubClient.Credentials = new Credentials(token);
+Cmd("git", $"checkout -b {releaseBranchName}");
+Cmd("git", $"push {remote} {releaseBranchName}");
+
+Console.WriteLine($"Creating pull request to merge {releaseBranchName} into master...");
+var pr = await gitHubClient.PullRequest.Create(
+    repoOwner,
+    repoName,
+    new NewPullRequest($"Release {version}", releaseBranchName, "master"));
+Console.WriteLine($"Created pull request {pr.Number}: '{pr.Title}'");
+
+
+public static string ReadTokenFromNetrc()
+{
+    const string githubApiHost = "api.github.com";
+    var netrcContents = File.ReadAllText(Path.Combine(Environment.GetEnvironmentVariable("HOME"), ".netrc"));
+    var tokens = netrcContents.Split(new[] { ' ', '\t', '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+
+    int lastTokenToCheck = tokens.Length - 1;
+    int i = 0;
+    for (; i < lastTokenToCheck; i += 2)
+    {
+        if (tokens[i] == "machine" && tokens[i + 1] == githubApiHost)
+        {
+            break;
+        }
+    }
+
+    for (i += 2; i < lastTokenToCheck; i += 2)
+    {
+        if (tokens[i] == "password")
+        {
+            return tokens[i + 1];
+        }
+
+        if (tokens[i] == "machine" || tokens[i] == "default")
+        {
+            break;
+        }
+    }

```

Comment:
The format is quite ugly... Is it worth sticking to the `.netrc` format? We could just have a `githubtoken` file in the repo that contains just the token (of course it would have to be added to `.gitignore`).

Reply:
We can proceed as you suggest. I only conformed to the `.netrc` format because it's established and tools like Octokit.rb use it and maybe Octokit.NET will eventually, but I'm not married to it. I would not mind seeing this method go away!

Reply:
Something else. Having one file in the repo means that we'd have to do a similar thing for _every repo_ that we set up like this. Which is probably fine. At least for now.

Reply:
True. If it comes to this, we could also look for the file in parent folders, like `Directory.Build.props`